tests/cases/standalone/common/promql/label.sql

```diff
@@ -52,6 +52,23 @@ TQL EVAL (0, 15, '5s') label_replace(test{host="host2"}, "idc", "$2", "idc", "(.
 -- SQLNESS SORT_RESULT 3 1
 TQL EVAL (0, 15, '5s') label_replace(test{host="host2"}, "idc", "", "", "");
 
+-- Issue 5726 --
+-- SQLNESS SORT_RESULT 3 1
+TQL EVAL(0, 15, '5s') label_replace(vector(1), "host", "host1", "", "");
+
+-- SQLNESS SORT_RESULT 3 1
+TQL EVAL(0, 15, '5s') {__name__="test",host="host1"} * label_replace(vector(1), "host", "host1", "", "");
```

Comment:
Add new cases:

```
TQL EVAL(0, 15, '5s') {__name__="test",host="host1"} * label_replace(vector(1), "host", "host2", "host", "");
```

```
TQL EVAL(0, 15, '5s') {__name__="test",host="host1"} * label_replace(vector(1), "host2", "host1", "instance", "");
```

Reply:
Added. But they are not addressed in this PR. It's not the same issue with #5726 or #6738.

To address them, `label_replace` requires special handling for the `vector` function.

Reply:
Try my best to fix it in https://github.com/GreptimeTeam/greptimedb/pull/6443/commits/ab960ecf5fbaf332428a0f7300fb3ba53532b65a